cmake/modules/CMakeLists.txt

```diff
@@ -1,5 +1,3 @@
-if(PROJECT_IS_TOP_LEVEL)
-  export(EXPORT SwiftSyntaxTargets
-         FILE ${CMAKE_CURRENT_BINARY_DIR}/SwiftSyntaxConfig.cmake
-         NAMESPACE SwiftSyntax::)
-endif()
+export(EXPORT SwiftSyntaxTargets
+       FILE ${CMAKE_CURRENT_BINARY_DIR}/SwiftSyntaxConfig.cmake
+       NAMESPACE SwiftSyntax::)
```

Comment:
I feel `SwiftSyntaxConfig.cmake` should be exported from swift repo's CMakeLists.txt.
Could you add `EXPORT` to 
https://github.com/apple/swift/blob/3094e661f51c9190ba9da2c3ef60cff80e28fa8d/lib/CMakeLists.txt#L32-L35
https://github.com/apple/swift/blob/3094e661f51c9190ba9da2c3ef60cff80e28fa8d/lib/CMakeLists.txt#L60-L66
and put `export(EXPORT)` there?

Reply:
I don't think that works. We are using `FetchContent` rather than `add_subdirectory` and as a result, this is a fully isolated build and thus the `install` rules are not evaluated as part of the Swift repository.  I don't think that trying to reproduce something like this by hand makes sense.

Reply:
I tried this patch https://github.com/rintaro/swift/commit/2a6d5b3a2b10ebca9a572486f59beb4e6f33cf72 and `${SWIFT_BUILD_DIR}/cmake/modules/SwiftSyntaxConfig.cmake` seems to be contain the correct information.

Reply:
Ah, that is a clever idea!  I think that I can make this work, and I assume that you are okay with installing the swiftmodules and import libraries as well from the Swift toolchain?

Reply:
Seems that I was wrong! I was mixing up the `FetchContent_MakeAvailable` and `FetchContent_Provide`.  So, I think that you are right we can just hoist the export.  Thanks for the idea!

Reply:
Yeap, this works to build!  Just verified locally.